src/endpoints/borrowed/delete/delete_by_id.py

```diff
@@ -42,6 +43,54 @@ async def delete_borrowed(
         raise custom_exception(
             status_code=status.HTTP_404_NOT_FOUND, details="Borrowed not found."
         )
+    found_copy = db.scalar(
+        select(all_models.Copy).where(
+            and_(
+                all_models.Copy.id == found_borrowed.copy_id,
+                not_(all_models.Copy.is_deleted),
+            )
+        )
+    )
+    if not found_copy:
+        logging.warning(
+            "Copy not found in database with id: " + str(found_borrowed.copy_id)
+        )
+        raise custom_exception(
+            status_code=status.HTTP_404_NOT_FOUND, details="Copy not found."
+        )
+    found_status = db.scalar(
+        select(all_models.Status).where(
+            and_(
+                all_models.Status.id == found_copy.status_id,
+                not_(all_models.Status.is_deleted),
+            )
+        )
+    )
+    if not found_status:
+        logging.warning(
+            "Status not found in database with id: " + str(found_copy.status_id)
+        )
+        raise custom_exception(
+            status_code=status.HTTP_404_NOT_FOUND, details="Status not found."
+        )
+    if found_status.status != BORROWED:
+        logging.warning("Status is not borrowed.")
+        raise custom_exception(
```

Comment:
It is not necessary that the status should be borrowed. It can be available as well (because the user returned the book), so this check is not needed here.